pkg/endpoint/regenerationcontext.go

```diff
@@ -21,12 +21,43 @@ import (
 	"github.com/cilium/cilium/pkg/revert"
 )
 
+// DatapathRegenerationLevel determines what is expected of the datapath when
+// a regeneration event is processed.
+type DatapathRegenerationLevel int
+
+const (
+	// RegenerateWithoutDatapath indicates that datapath rebuild or reload
+	// is not required to implement this regeneration.
+	RegenerateWithoutDatapath DatapathRegenerationLevel = 0
```

Comment:
you could use [Iota](https://github.com/golang/go/wiki/Iota) here if you want

Reply:
I tried, for some reason (maybe because they're typed with a non-int type?) the compiler disagreed with me. So I just set them manually.